Sources/Sentry/SentryNSDataTracker.m

```diff
@@ -173,9 +179,32 @@ - (NSData *)measureNSDataFromURL:(NSURL *)url
 
     [ioSpan setDataValue:path forKey:@"file.path"];
 
+    [self mainThreadExtraInfo:ioSpan];
+
     return ioSpan;
 }
 
+- (void)mainThreadExtraInfo:(id<SentrySpan>)span
+{
+    BOOL isMainThread = [NSThread isMainThread];
+
+    [span setDataValue:@(isMainThread) forKey:@"blocked_main_thread"];
+
+    if (!isMainThread) {
+        return;
+    }
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
+    SentryStacktrace *stackTrace = [threadInspector stacktraceForCurrentThreadNatively];
+
+    NSMutableArray *frames = [[NSMutableArray alloc] initWithCapacity:stackTrace.frames.count];
+    for (SentryFrame *frame in stackTrace.frames) {
+        [frames addObject:[frame serialize]];
```

Comment:
`m`: I'm a bit worried about this piece of code. If you have many file IO operations on the main thread, this can drastically slow you down and increase the payload. We could, for example, only include the 20 topmost frames, which should be enough for you to locate the problem and enough for grouping. I started a thread in [Slack](https://sentry.slack.com/archives/G01PW6HRQNN/p1672825886909859) to discuss thi.

Reply:
The original solution was to include only frames from the app itself. An idea that I really like. I thought this would not be possible because I would have to remove local symbolization. Now that I refactor the symbolization process and is fast enough I think we can go back to this original idea

Reply:
Let's not do this for now, as we only serialize if you are doing file IO on the main thread, and it's acceptable to slow down applications a bit to notify them that they are doing it on the main thread.